From 3aec38a84ac850c00944d2f2715d0302b0f2399f Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Tue, 13 Mar 2018 17:11:57 +0800
Subject: [PATCH 3492/5242] MLK-17815-1 dts: arm64: imx8qm: add sata phy
 region

commit  00d85eadb2a2f4f350b3bd54cddbdd3263829fec from
https://source.codeaurora.org/external/imx/linux-imx.git

Add the extra imx8qm sata phy register region,
and the clock phy_apbclk, mandatory required to
access phy registers.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 1403df3..585eb52 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -3637,16 +3637,19 @@
 
 	sata: sata@5f020000 {
 		compatible = "fsl,imx8qm-ahci";
-		reg = <0x0 0x5f020000 0x0 0x10000>; /* Controller reg */
+		reg = <0x0 0x5f020000 0x0 0x10000>, /* Controller reg */
+			<0x0 0x5f1a0000 0x0 0x10000>; /* PHY reg */
+		reg-names = "ctl", "phy";
 		interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&clk IMX8QM_HSIO_SATA_CLK>,
 			 <&clk IMX8QM_HSIO_PHY_X1_PCLK>,
 			 <&clk IMX8QM_HSIO_SATA_EPCS_TX_CLK>,
 			 <&clk IMX8QM_HSIO_SATA_EPCS_RX_CLK>,
 			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_0>,
-			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_1>;
+			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_1>,
+			 <&clk IMX8QM_HSIO_PHY_X1_APB_CLK>;
 		clock-names = "sata", "sata_ref", "epcs_tx", "epcs_rx",
-				"phy_pclk0", "phy_pclk1";
+				"phy_pclk0", "phy_pclk1", "phy_apbclk";
 		hsio = <&hsio>;
 		power-domains = <&pd_sata0>;
 		iommus = <&smmu 0x13 0x7f80>;
-- 
1.7.9.5

